### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by chiztechnology**
Hello! We just set up a new feature to Preserve the data from the form using localStorage and store them in the browser.
Kindly check and tell us if there's something to improve.
Thanks a lot 🙇‍♂️

Faithfully

### CODE REVIEW

File: js/form_validator.js [hunk 1]
```diff
@@ -49,7 +49,24 @@ const EMAIL_INVALID = 'Please enter a correct email address format !';
...
+    formData.append('email', form.elements.email.value);
+    formData.append('message', form.elements.message.value);
+    formData.forEach((value, key) => {
+      myObject[key] = value;
+    });
+    localStorage.setItem('form-data', JSON.stringify(myObject));
   }
-});
\ No newline at end of file
+});
```
Comment (1GORDON) [inline js/form_validator.js]: Great job @chiztechnology 🤝

- [ ] As stated in the project requirements, kindly make sure input data persists in the browser and is saved to the local storage. At the moment input data is not saved to the local storage.
Empty

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @chiztechnology ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/form_validator.js (hunk 1 above)
Comment (chiztechnology) [reply js/form_validator.js]: dear @1GORDON  everything is working!
Perfectly working. I don't find anything to update !
please check twice and let me know

Comment (1GORDON) [reply js/form_validator.js]: Hi @chiztechnology. On my side, I could not see data being saved in the local storage. If everything is working well then you may resubmit and the next reviewer will approve your project.

Comment (chiztechnology) [reply js/form_validator.js]: @1GORDON please kindly check this sceen recording video.
https://drive.google.com/file/d/11fdw3Yp_JqvUZ4RQNs3oCfvY_B6MVxJ6/view?usp=sharing

Comment (chiztechnology) [reply js/form_validator.js]: make sure you're running from the appropriate branch

File: js/form_validator.js [hunk 2]
```diff
@@ -49,7 +49,24 @@ const EMAIL_INVALID = 'Please enter a correct email address format !';
...
   if (!emailValid) {
     event.preventDefault();
+  } else {
+    formData.append('name', form.elements.name.value);
+    formData.append('email', form.elements.email.value);
+    formData.append('message', form.elements.message.value);
+    formData.forEach((value, key) => {
+      myObject[key] = value;
+    });
+    localStorage.setItem('form-data', JSON.stringify(myObject));
```
Comment (BenMukebo) [inline js/form_validator.js:102]: - [ ] Great job so far however, as requested by the previous reviewer, When the user changes the content of any input field in the form, the data object must be updated and saved to the local storage. even without having to submit the form, When the user loads the page, the input fields must be pre-filled with the data. kindly add this 👍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @chiztechnology, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- No linter errors. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️

 ## Required Changes ♻️
_Check the comments under the review._

You have submitted a project, that Linters check are not running in the GitHub/action
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/form_validator.js (hunk 2 above)
Comment (chiztechnology) [reply js/form_validator.js:102]: thank you! I resolved that.

**Review by Santiago220991 — APPROVED** [review]
Hi @chiztechnology, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
:heavy_check_mark: Applied suggested by the last code reviewer.
:heavy_check_mark: No linter errors.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**